src/VecSim/spaces/IP_space.cpp

```diff
@@ -25,7 +36,7 @@ InnerProductSpace::InnerProductSpace(size_t dim, std::shared_ptr<VecSimAllocator
             fstdistfunc_ = InnerProductSIMD16Ext_AVX;
         } else if (dim % 4 == 0) {
             fstdistfunc_ = InnerProductSIMD4Ext_AVX;
-        } else if (dim > 16) {
+        } else if (dim > 16 && dim % 16 < 4) {
```

Comment:
Why lines 39 and 52 are not covered?

Reply:
making a test for every available optimization is complicated. you have to run the test on different machines to go through all lines here. we test only the functions themself and not the logic.
the only lines that are covered are the lines the coverage test machine is directed to through our other tests